ring/model.go

```diff
@@ -192,8 +194,11 @@ func (d *Desc) mergeWithTime(mergeable memberlist.Mergeable, localCAS bool, now
 		return nil, nil
 	}
 
-	thisIngesterMap := buildNormalizedIngestersMap(d)
-	otherIngesterMap := buildNormalizedIngestersMap(other)
+	normalizeIngestersMap(d)
+	normalizeIngestersMap(other)
```

Comment:
Nit: I would suggest to make `Normalize` a method on `(d *Desc)`, and make comment say that `Merge calls Normalize on mergeable parameter.` Right now the comment says that "This method modifies ... mergeable", but random modification would break commutativity and associativity. Being specific about normalization explains that this isn't breaking any rules of Mergeable interface.

Reason for having `Normalize` method is to document that it only "cleans up" the receiver, but doesn't change it's meaning.

Reply:
Bear in mind that from a caller perspective, the effect of Merge on Desc does not change, it was normalized in-place previously. What is different, is that when a change is not detected, the normalization still occurs. 

I think this is a good idea, if we mandate that callers are responsible for calling `Normalize`; I'm not convinced to make it public unnecessarily. I'd like to leave this as-is because I am working on another change that (I think) clarifies the Mergeable interface, and give us more opportunities to optimize.

Reply:
👍 I'm fine keeping as-is, with above-suggested clarification of Mergeable interface.